### DISCUSSION TITLE
StyledProperty doing nothing

### DISCUSSION THREAD

**Discussion by Abdesol** [AvaloniaUI/Avalonia#7081] [Q&A] [answered]
I created a custom combo box and it has some properties registered. All of them are working except the SelectedIndex property
The below code is for SelectedIndex property:
```cs
public new static readonly StyledProperty<int> SelectedIndexProperty =
    AvaloniaProperty.Register<ComboBox, int>(nameof(SelectedIndex));
public int SelectedIndex
{
    get => GetValue(SelectedIndexProperty);
    set => SetValue(SelectedIndexProperty, value);
}

private static void SelectedIndexPropertyChanged(AvaloniaPropertyChangedEventArgs<int> e)
{
    if (e.Sender is CustomComboBox ctrl)
    {
        ctrl.comboBox.SelectedIndex = e.NewValue.Value;
    }
}
```
All my properties are done in a similar way but I don't know why this one is not working
And also, the change is subscribed to a method too.
```cs
 SelectedIndexProperty.Changed.Subscribe(SelectedIndexPropertyChanged);
```
I am using it in my view like the one below
```xml
<controls:CustomComboBox Grid.Column="0" Name="Sort by"
	 Items="{Binding Sorts}"
	 ItemSelected="{Binding ComboBoxCommand}"
	 SelectedIndex="0">
```
It should have been working.

Is there any way to solve it?
Thanks in advance  : )

**Comment by maxkatz6** [comment, marked as answer]
@simonhaines I don't see any issue with this code using integer (as I saw you solved it with string, which indeed a weird way).
Only one problem that I definitely see - you did not provide any default value. So it uses "0" as a default. Which means if you set value to 0, there won't be any change in fact. It might work with string because default will be "null".

So try to set default to "-1" with integer.

Also we don't see whole CustomComboBox control in case if there any other problem.

**Reply by Abdesol** [reply]
I understood the problem and indeed it worked as I expected it.
I am from a WPF background, that's why I messed up with it.
